js/main.js

```diff
@@ -170,7 +180,39 @@ $ulElement.addEventListener('click', function (event) {
         $notesBox.value = data.editing.notesKey;
 
         $editH2.textContent = 'Edit Entry';
+
+        $changeContentRow.setAttribute('class', 'row column-full space-between');
+        $deleteEntryButton.setAttribute('class', '');
       }
     }
   }
 });
+
+$deleteEntryButton.addEventListener('click', function (event) {
+  $modalBg.setAttribute('class', 'modal-bg');
+});
+
+$cancelButton.addEventListener('click', function (event) {
+  $modalBg.setAttribute('class', 'modal-bg hidden');
+});
+
+$confirmButton.addEventListener('click', function (event) {
+  for (var i = 0; i < data.entries.length; i++) {
+    if (data.entries[i].entryID === data.editing.entryID) {
+      var dataobjIndex = data.entries.indexOf(data.entries[i]);
+      data.entries.splice(dataobjIndex, 1);
+    }
+  }
+  var $liInUl = $ulElement.querySelectorAll('li');
+  for (var j = 0; j < $liInUl.length; j++) {
+    if (parseInt($liInUl[j].getAttribute('data-entry-id')) === data.editing.entryID) {
+      $liInUl[j].remove();
+    }
+  }
+
+  if (data.entries.length === 0) {
+    toggleNoEntries();
+  }
+  $modalBg.setAttribute('class', 'modal-bg hidden');
+  viewSwap('entries');
```

Comment:
You will need to decrement data.entries.entryId by one. If you look in your gif you will see that this number keeps growing as you delete items and add items.